splunk-otel-android/src/test/java/io/opentelemetry/rum/internal/instrumentation/anr/AnrDetectorTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.rum.internal.instrumentation.anr;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor.constant;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import android.os.Looper;
+import io.opentelemetry.rum.internal.instrumentation.InstrumentedApplication;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AnrDetectorTest {
+
+    @Mock Looper mainLooper;
+    @Mock ScheduledExecutorService scheduler;
+    @Mock InstrumentedApplication instrumentedApplication;
+
+    @Captor ArgumentCaptor<AnrWatcher> anrWatcherCaptor;
+
+    @Test
+    void shouldInstallInstrumentation() {
+        when(instrumentedApplication.getOpenTelemetrySdk())
+                .thenReturn(OpenTelemetrySdk.builder().build());
+
+        AnrDetector anrDetector =
+                AnrDetector.builder()
+                        .setMainLooper(mainLooper)
+                        .setScheduler(scheduler)
+                        .addAttributesExtractor(constant(stringKey("test.key"), "abc"))
+                        .build();
+        anrDetector.installOn(instrumentedApplication);
+
+        // verify that the ANR scheduler was started
+        verify(scheduler)
+                .scheduleAtFixedRate(
+                        anrWatcherCaptor.capture(), eq(1L), eq(1L), eq(TimeUnit.SECONDS));
```

Comment:
If we don't care about the captured value, isn't it better just to use `isA(AnrWatcher.class)`?

Reply:
Oops, that was a leftover of my experiments with these tests. Removed!